authorize.go

```diff
@@ -19,6 +19,9 @@ const (
 // IsAuthorized uses the combination of the PolicySet and Entities to determine
 // if the given Request to determine Decision and Diagnostic.
 func (p PolicySet) IsAuthorized(entities types.EntityGetter, req Request) (Decision, Diagnostic) {
+	if entities == nil {
+		entities = types.EntityMap{}
+	}
```

Comment:
Oh Golang...

Reply:
Lol, I think ... slightly better might be (again, this is Go):

```
var zero types.EntityMap
entities = zero
```

Which is because (and feel free to Go playground / check on this explanation) when you just say `var zero types.EntityMap` you are yielding what amounts to a typed nil pointer.  However, when you do `entities = types.EntityMap{}` you are allocating an empty map.

In our case, probably either would work fine, but the `var zero` more precisely replicates the original behavior and avoids an unnecessary allocation.

Reply:
Agreed with Phil on avoiding the extra allocation, as nitpicky as it may be.

I'm not sure who would ever pass a nil `EntityGetter`.